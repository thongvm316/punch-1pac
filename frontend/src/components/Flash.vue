<template>
  <transition name="fade">
    <div class="toast toast-success text-center toast-flash"
      :class="{ 'toast-success': type === 'success', 'toast-error': type === 'error' }"
      v-if="message">
      {{ message }}
    </div>
  </transition>
</template>

<script>
import { SET_FLASH_MESSAGE } from '../store/mutation-types'
import { mapState, mapMutations } from 'vuex'

export default {
  name: 'flash',

  computed: {
    ...mapState('flash', ['message', 'type', 'timeout'])
  },

  methods: {
    ...mapMutations('flash', [SET_FLASH_MESSAGE])
  },

  watch: {
    message: function(newMsg) {
      if (newMsg) {
        setTimeout(() => {
          this[SET_FLASH_MESSAGE]({ message: null })
        }, this.timeout)
      }
    }
  }
}
</script>
