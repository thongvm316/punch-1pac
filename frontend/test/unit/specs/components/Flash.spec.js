import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import setComputed from '../../supports/set-computed'
import Flash from '@/components/Flash'

const localWrapperOps = {
  ...wrapperOps,
  computed: {
    message: () => 'Success Message',
    type: () => 'success'
  }
}
const SET_FLASH_MESSAGE = jest.spyOn(Flash.methods, 'SET_FLASH_MESSAGE')

describe('Flash.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Flash, localWrapperOps)
  })

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when Flash was mounted', () => {
    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })

    it('should render error message', async () => {
      setComputed(wrapper, {
        message: 'Error Message', type: 'error'
      })
      await wrapper.vm.$nextTick()

      expect(wrapper).toMatchSnapshot()
    })
  })
})
