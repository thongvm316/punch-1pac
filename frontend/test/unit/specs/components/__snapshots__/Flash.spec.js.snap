// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flash.vue when Flash was mounted should render correctly 1`] = `
<transition-stub name="fade">
  <div class="toast toast-success text-center toast-flash toast-success">
    Success Message
  </div>
</transition-stub>
`;

exports[`Flash.vue when Flash was mounted should render error message 1`] = `
<transition-stub name="fade">
  <div class="toast toast-success text-center toast-flash toast-error">
    Error Message
  </div>
</transition-stub>
`;
